tools/scripts/generic.mk

```diff
@@ -333,11 +346,21 @@ ifneq ($(strip $(LSCRIPT)),)
 LSCRIPT_FLAG = -T$(LSCRIPT)
 endif
 
+ifeq 'mbed' '$(PLATFORM)'
+$(BINARY): $(OBJS) $(LSCRIPT) $(MBED_OS_LIBRARY)
+	@$(call print,[LD] $(notdir $(OBJS)))
+	$(MUTE) $(call print, putting project object files names to text file)
+	$(MUTE) $(foreach file,$(sort $(OBJS)),$(call APPEND_TO_FILE,$(file),$(OBJECT_FILES_NAMES)) $(cmd_separator)) echo . $(HIDE)
+	$(MUTE) cd $(MBED_OS_DIRECTORY) $(cmd_separator) $(CC) $(LSCRIPT_FLAG) $(LDFLAGS) -o $(BINARY) @$(OBJECT_FILES_NAMES) @$(MBED_GENERATED_ARCHIVE_FILE)\
+			$(LIB_FLAGS)
+	$(MUTE) $(MAKE) --no-print-directory post_build
+else
```

Comment:
I think this should go in mbed.mk
We shouldn't have these type of switches in the generic.mk file

Reply:
As for mbed platform number of objects files are more and in windows it is giving character limit issue. So i copied all the object files names to text file and used in this linking stage. I can't put these rule in mbed.mk, as if i put it in mbed.mk there will the two rules for same target(BINARY).

Reply:
@rbolboac is right, there must be a way to move this in mbed.mk

Reply:
When we run make command it will run update and then build rule, For build target, prerequisite is BINARY so we can't use any other variable here. So if we put a rule with target same as BINARY in med.mk then there will be two recipe for one target which is not allowed in makefile.
What we propose is, we need to copy all object files name to a text file and use in linking stage. As we are doing this to overcome the character limit issues in window's. So we can adopt this method here also to avoid this issues appearing in future for other platform in window's. This approach will not break any existing build method. So linking stage recipe will look something like below,
$(MUTE) $(CC) $(LSCRIPT_FLAG) $(LDFLAGS) $(LIB_PATHS) -o $(BINARY) @$(OBJECT_FILES_NAMES) $(EXTRA_FILE_NAME) $(BOOTOBJ) $(ASM_OBJS) $(LIB_FLAGS)  
So by using above mentioned modification we can remove platform checking condition here.

Reply:
Hi Team,

Is this change okay?

Reply:
When you say "Windows' character limit", are you referring to the maximum path length (260 characters) limitation?

Reply:
It is the single make file command limit i guess. Because linker stage require all the object files names linker command becomes much long, So it will be giving errors.
I tried to echo all the object files names but when i do that, echo will not print all the object names.

Reply:
So @CMinajigi I still don't understand why it's not possible to move this stuff in mbed.mk.
The only thing that's different is that you do some steps (print a message and generate that list of files) before actual build. To achieve this by moving all this in mbed.mk, create a pre_build step, like the existing post_build step. generic.mk invokes the pre_build step, then the linking, then the post_build step. This should be doable.